pkg/compactor/block_upload.go

```diff
@@ -365,6 +409,69 @@ func (c *MultitenantCompactor) uploadMeta(ctx context.Context, logger log.Logger
 	return nil
 }
 
+func (c *MultitenantCompactor) validateBlock(ctx context.Context, blockID ulid.ULID,
+	userBkt objstore.Bucket, meta metadata.Meta) error {
+
+	// create a temporary directory to download the block
+	// on normal compactors this will be in the compartor's data directory, but in other cases (like some tests)
+	// the fallback will be the system's temp directory
+	tmpDir := os.TempDir()
+	if _, err := os.Stat(c.compactorCfg.DataDir); !os.IsNotExist(err) {
+		tmpDir = c.compactorCfg.DataDir
+	}
+	blockDir, err := os.MkdirTemp(tmpDir, "upload")
+
+	level.Debug(c.logger).Log("msg", "created temporary block directory", "dir", blockDir)
+	if err != nil {
+		return errors.Wrap(err, "failed to create temp dir")
+	}
```

Comment:
Shouldn't the error be checked before logging that `blockDir` was created?
```suggestion
	blockDir, err := os.MkdirTemp(tmpDir, "upload")
	if err != nil {
		return errors.Wrap(err, "failed to create temp dir")
	}

	level.Debug(c.logger).Log("msg", "created temporary block directory", "dir", blockDir)
```